Flaky test: ProxyProtocolIT fails with 'Address already in use'

## Description

The `ProxyProtocolIT` test is flaky and intermittently fails in CI with port binding errors.

## Failure Details

- **Test class**: `io.kroxylicious.it.ProxyProtocolIT`
- **Error**: `io.netty.channel.unix.Errors$NativeIoException: bind(..) failed with error(-98): Address already in use`
- **Result**: `Tests run: 8, Failures: 0, Errors: 4, Skipped: 0, Time elapsed: 7.488 s <<< FAILURE!`

## CI Run

Failed in workflow run: https://github.com/kroxylicious/kroxylicious/actions/runs/24927007602/job/72998636894?pr=3797

## Stack Trace

```
java.util.concurrent.CompletionException: java.lang.RuntimeException: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed with error(-98): Address already in use
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:994)
	at java.base/java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(CompletableFuture.java:974)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2194)
	at io.kroxylicious.proxy.internal.net.NetworkBindRequest.lambda$performBindingOperation$0(NetworkBindRequest.java:72)
Caused by: java.lang.RuntimeException: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed with error(-98): Address already in use
	at io.kroxylicious.proxy.internal.net.EndpointRegistry.lambda$registerBinding$22(EndpointRegistry.java:459)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:990)
Caused by: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed with error(-98): Address already in use
```

## Analysis

This appears to be a port reuse/cleanup issue where:
1. A previous test or test instance hasn't fully released a port
2. The test tries to bind to a port that's still in use
3. Multiple bind attempts fail with the same error

This is a classic flaky test symptom in integration tests that use network ports.

## Possible Causes

- Insufficient wait time for port release after test cleanup
- Race condition in parallel test execution
- Port range exhaustion when multiple tests run concurrently
- Missing or incomplete cleanup in `@AfterEach`/`@AfterAll` hooks

## Suggested Fixes

1. Review port allocation strategy in test setup
2. Ensure proper cleanup of network resources
3. Add retry logic or wait time for port availability
4. Consider using dynamic port allocation (port 0) if not already in use
5. Review test isolation and ensure tests clean up properly

--- a/kroxylicious-integration-test-support/src/main/java/io/kroxylicious/test/tester/KroxyliciousConfigUtils.java
+++ b/kroxylicious-integration-test-support/src/main/java/io/kroxylicious/test/tester/KroxyliciousConfigUtils.java
@@ -28,7 +28,7 @@ public class KroxyliciousConfigUtils {
     public static final String DEFAULT_VIRTUAL_CLUSTER = "demo";
     public static final String DEFAULT_GATEWAY_NAME = "default";
 
-    static final HostPort DEFAULT_PROXY_BOOTSTRAP = new HostPort("localhost", 9192);
+    public static final HostPort DEFAULT_PROXY_BOOTSTRAP = new HostPort("localhost", 9192);
 
     /**
      * Create a KroxyliciousConfigBuilder with a single virtual cluster configured to

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/ProxyProtocolIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/ProxyProtocolIT.java
@@ -39,7 +39,6 @@ import io.kroxylicious.proxy.config.ConfigurationBuilder;
 import io.kroxylicious.proxy.config.ProxyProtocolConfig;
 import io.kroxylicious.proxy.config.ProxyProtocolMode;
 import io.kroxylicious.proxy.config.VirtualClusterBuilder;
-import io.kroxylicious.proxy.service.HostPort;
 import io.kroxylicious.proxy.tls.CertificateGenerator;
 import io.kroxylicious.test.Request;
 import io.kroxylicious.test.Response;
@@ -50,6 +49,7 @@ import io.kroxylicious.test.codec.KafkaRequestEncoder;
 import io.kroxylicious.test.codec.KafkaResponseDecoder;
 import io.kroxylicious.test.tester.KroxyliciousConfigUtils;
 
+import static io.kroxylicious.test.tester.KroxyliciousConfigUtils.DEFAULT_PROXY_BOOTSTRAP;
 import static io.kroxylicious.test.tester.KroxyliciousConfigUtils.defaultPortIdentifiesNodeGatewayBuilder;
 import static io.kroxylicious.test.tester.KroxyliciousTesters.mockKafkaKroxyliciousTester;
 import static org.assertj.core.api.Assertions.assertThat;
@@ -401,7 +401,7 @@ class ProxyProtocolIT {
                         .withNewTargetCluster()
                         .withBootstrapServers(mockBootstrap)
                         .endTargetCluster()
-                        .addToGateways(defaultPortIdentifiesNodeGatewayBuilder(new HostPort("localhost", 49592))
+                        .addToGateways(defaultPortIdentifiesNodeGatewayBuilder(DEFAULT_PROXY_BOOTSTRAP)
                                 .withNewTls()
                                 .withNewKeyStoreKey()
                                 .withStoreFile(keystore.path().toString())

